packages/protocol/contracts/ACE/ACE.sol

```diff
@@ -184,38 +215,6 @@ contract ACE {
         }
     }
 
-    function createNoteRegistry(
```

Comment:
It would improve readability to make the logic surrounding note registries a discrete contract, that `ACE.sol` inherits from. I don't think any of the logic needs to change, creating a discrete contract means we can approach testing ACE in a more systematic manner.  
  
Regarding the structure of an inherited `NoteRegistry.sol`, this base contract needs to validate the legitimacy of proofs via `validateProofByHash`. This should be a virtual function inside `NoteRegistry.sol` that `ACE.sol` overrides.